spinflip/matrix.py

```diff
@@ -55,15 +61,30 @@ def visualizer(M, log=True,  text='mag', traceless = True, vmin=1E-15,vmax=1E-6,
             for m in np.arange(0,matrix_size):
                 xcoord = (n+1/6)*grid_subdivisions
                 ycoord = 1 - (m+1/2)*grid_subdivisions
-                ax.text(xcoord, ycoord, str(round(np.log10(np.abs(M[m,n])),0)), color='cyan', size=10)
+                if log == True:
+                    ax.text(xcoord, ycoord, str(round(np.log10(np.abs(M[m,n])),0)), color='cyan', size=10)
+                else:
+                    ax.text(xcoord, ycoord, str(round(np.abs(M[m,n]),2)), color='cyan', size=10)
                 
     elif text=='arg':
         for n in np.arange(0,matrix_size):
             for m in np.arange(0,matrix_size):
                 xcoord = (n)*grid_subdivisions
                 ycoord = 1 - (m+1/2)*grid_subdivisions
                 ax.text(xcoord, ycoord, str(round(np.real(M[m,n]/np.abs(M[m,n])), ndigits=1))+'+'+str(round(np.imag(M[m,n])/np.abs(M[m,n]), ndigits=2))+'i', color='cyan', size=9)
-
+    elif text == 'full':
+        for n in np.arange(0,matrix_size):
+            for m in np.arange(0,matrix_size):
+                xcoord = (n)*grid_subdivisions
+                ycoord = 1 - (m+1/2)*grid_subdivisions
+                ax.text(xcoord, ycoord, str(round(np.real(M[m,n]), ndigits=2))+'+'+str(round(np.imag(M[m,n]), ndigits=2))+'i', color='cyan', size=9)
+    f.show()
     if savefig == True: 
         plt.tight_layout()
         plt.savefig('visualizer.png', dpi=300)
+
+def derivative(H,P, n=1):
```

Comment:
I think this would be better named something like `dState_dt`. Also comment what the function does and what the arguments mean.